main/init/test/src/mill/init/maven/InitModuleTests.scala

```diff
@@ -0,0 +1,134 @@
+package mill.init.maven
+
+import mill.testkit.{TestBaseModule, UnitTester}
+import utest.*
+
+object InitModuleTests extends TestSuite {
+
+  def tests: Tests = Tests {
+
+    val sourceRoot: os.Path = os.Path(sys.env("MILL_TEST_RESOURCE_DIR")) / "maven"
+
+    object initmodule extends TestBaseModule with InitModule
+
+    test("single") {
+
+      val eval = UnitTester(initmodule, sourceRoot / "single-module")
+
+      eval(initmodule.init()).isRight ==> true
+
+      os.walk(initmodule.millSourcePath).count(_.ext == "mill") ==> 1
+      os.exists(initmodule.millSourcePath / "build.mill") ==> true
+
+      val buildContents = os.read(initmodule.millSourcePath / "build.mill")
+
+      buildContents.contains(
+        s"""  override def pomSettings = PomSettings(
+           |    s\"\"\"Sample single module Maven project with a working, deployable site.\"\"\".stripMargin,
+           |    "",
+           |    "http://www.example.com",
+           |    Seq(),
+           |    VersionControl(
+           |      Option("http://github.com/gabrielf/maven-samples"),
+           |      Option("scm:git:git@github.com:gabrielf/maven-samples.git"),
+           |      Option("scm:git:git@github.com:gabrielf/maven-samples.git"),
+           |      Option("HEAD")
+           |    ),
+           |    Seq()
+           |  )
```

Comment:
Let's go all the way and use a golden file on disk that we use for comparisons. That way it's much clearer what is being compared (v.s. this long list of) and it can still be easily updated by just copying the generated file over the golden file